packages/ra-ui-materialui/src/field/BooleanField.tsx

```diff
@@ -85,11 +88,14 @@ BooleanField.propTypes = {
     looseValue: PropTypes.bool,
 };
 
+BooleanField.propTypes = BooleanFieldImpl.propTypes;
```

Comment:
Same, please define the proptypes of the implementation first then copy it to the memoized component. By the way, can't you bake in this proptype copy in the `genericMemo` component? Same for the `displayName` copy.